src/autogpt_plugins/news_search/test_auto_gpt_news_search.py

```diff
@@ -0,0 +1,40 @@
+import os
+import requests
+import unittest
+from typing import List
+from . import AutoGPTNewsSearch
+from .news_search import news_search
+
+
+class TestAutoGPTNewsSearch(unittest.TestCase):
+    def setUp(self):
+        os.environ["NEWSAPI_API_KEY"] = "test_key"
+        self.plugin = AutoGPTNewsSearch()
+
+    def tearDown(self):
+        os.environ.pop("NEWSAPI_API_KEY", None)
+
+    def test_news_search(self):
+        query = "test query"
+        try:
+            news_search(query)
+        except requests.exceptions.HTTPError as e:
+            self.assertEqual(e.response.status_code, 401)
```

Comment:
We don't need to test their api works, so use a pytest mock to make sure it can return data. Take a look at the main repo for how we are doing that with openai for inspiration

Reply:
Added a simple patch. I had copied this version from the bing_search plugin, although you are right this isn't the proper way